include/spdlog/sinks/eventlog_sink.h

```diff
@@ -0,0 +1,100 @@
+// Copyright(c) 2015-present, Gabi Melman & spdlog contributors.
+// Distributed under the MIT License (http://opensource.org/licenses/MIT)
+
+// Requires the following registry entries to be present, with the following modifications:
+// 1. {app_name} should be replaced with your application name
+// 2. {log_name} should be replaced with the specific log name and the key should be duplicated for
+//               each log used in the application
+
+/*---------------------------------------------------------------------------------------
+
+Windows Registry Editor Version 5.00
+
+[HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\EventLog\{app_name}]
+
+[HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\EventLog\{app_name}\{log_name}]
+"TypesSupported"=dword:00000007
+"EventMessageFile"=hex(2):25,00,73,00,79,00,73,00,74,00,65,00,6d,00,72,00,6f,\
+  00,6f,00,74,00,25,00,5c,00,53,00,79,00,73,00,74,00,65,00,6d,00,33,00,32,00,\
+  5c,00,6d,00,73,00,63,00,6f,00,72,00,65,00,65,00,2e,00,64,00,6c,00,6c,00,00,\
+  00
+
+-----------------------------------------------------------------------------------------*/
+
+#pragma once
+
+#ifdef _WIN32
+#include <winbase.h>
+#endif
+
+#include <spdlog/details/null_mutex.h>
+#include <spdlog/sinks/base_sink.h>
+
+#include <mutex>
+#include <string>
+
+namespace spdlog {
+namespace sinks {
+/*
+ * Windows Event Log sink
+ */
+template<typename Mutex>
+class eventlog_sink : public base_sink<Mutex>
+    {
+    public:
+        explicit eventlog_sink(
+            std::string const& source,
+            std::string const& log = "Application",
+            std::string const& message_file_path = "%windir%\\System32\\mscoree.dll");

```

Comment:
why this dll in particualr ?

Reply:
It's a part of Microsoft .NET Framework that is installed on pretty much every Windows system out there, and this dll happens to contain the needed resource. I don't think this is going to change any time soon, but since it is still an undocumented hack, I also give a way to do that with a custom built dll if so desired or if Microsoft changes their old dll or whatever.

Reply:
Thus approach still seems bit confusing to me. If we expect the user to set those reg entries beforehand using regedit, then there is no real benefit for all this registry code, and it can be simplified dramatically. 
I suggest removing the registry code  it completely and let the user take care of it as written in your top comment (use regedit).